benchmarks/benchmarks/bench_core.py

```diff
@@ -161,6 +161,23 @@ def time_count_nonzero_multi_axis(self, numaxes, size, dtype):
                 self.x.ndim - 1, self.x.ndim - 2))
 
 
+class Nonzero(Benchmark):
+    params = [
+    [bool, np.uint8, np.int8, np.uint16, np.int16, np.uint32, np.int32, np.uint64, np.int64, np.float32, np.float64],
+    [(1000000,), (1000,1000), (100,100,100)]
+    ]
+    param_names = ["dtype", "shape"]
+
+    def setup(self, dtype, size):
+        self.x = np.random.randint(0,2,size=size).astype(dtype)
```

Comment:
We also need benchmarks for "sparse" and "nonsparse" data.

That is, please generate a dataset which has very few (maybe only a single 1 somewhere in the middle or at the end), and a dataset which has everything 1.

Reply:
It might also be good to seed this, just to stabilize benchmarks a bit (although they tend to fluctuate absurdly often anyway).